server/athenian/api/controllers/team_controller.py

```diff
@@ -305,6 +306,63 @@ async def get_team_from_db(
     return team
 
 
+async def teams_recursive_fetch(
+    account_id: int,
+    sdb_conn: DatabaseLike,
+    select_entities: Sequence[InstrumentedAttribute] = (Team.id,),
+    root_team_ids: Optional[Collection] = None,
+    max_depth: int = None,
+) -> Sequence[Mapping[Union[int, str], Any]]:
+    """Return the recursively collected list of teams for the account.
+
+    If `root_team_ids` is passed those will be taken as base for the recursion.
+    If `root_team_ids` is None all root teams will be taken as base for the recursion.
+
+    The returned list of teams will include duplicates when
+    on of the `root_team_ids` is ancestor of another.
+
+    Returned columns can be selected with `select_entities`. The ID of the root team
+    used to fetch the row will always be included as last column.
+
+    """
+    # a recursive CTE is used to link children teams and track depth
+
+    # Team.id is required inside cte for join even if not requested by caller
+    if Team.id not in select_entities:
+        cte_select_entities: Sequence[InstrumentedAttribute] = (*select_entities, Team.id)
+    else:
+        cte_select_entities = select_entities
+
+    # base team is the specified root_team_id if present, else all account's root teams
+    recursive_base_where = Team.owner_id == account_id
+    if root_team_ids is None:
+        recursive_base_where = sa.and_(recursive_base_where, Team.parent_id == None)  # noqa: E711
```

Comment:
```suggestion
        recursive_base_where = sa.and_(recursive_base_where, Team.parent_id.is_(None))
```

Reply:
nice, it was horrible to add the flake8 override